### PR TITLE
Add an overview vignette

### LINKED ISSUE #274: blocking/grouping documentation
It appears that the [rsampling docs](https://www.tidymodels.org/start/resampling/) don't contain info on blocking/grouping, as is described in the [mlr package docs](https://mlr.mlr-org.com/articles/tutorial/resample.html) or the [scikit-learn package docs](https://scikit-learn.org/stable/auto_examples/model_selection/plot_cv_indices.html#sphx-glr-auto-examples-model-selection-plot-cv-indices-py). It would be helpful to have that info in the "get started" docs, given how common the need for blocking/grouping is in ML.

### LINKED ISSUE #347: Expand documentation for different `balance` methods for grouped sampling
Hi @mikemahoney218,

There seems to be limited explanation anywhere (AFAIK) about how the different balancing methods work for grouped sampling. Even in `make_groups()`, the docs sort of describes how `balance = 'prop'` works, but I'm left unsure exactly what balancing by `'groups'` or `'observations'` means in this context. 

I could maybe make an educated guess, but I think some expanded documentation in `make_groups()` would be a welcome addition and improve clarity.

### PR DESCRIPTION
**Pull request by mikemahoney218**
This PR adds a vignette giving an overview of the main resampling functions in rsample, with brief descriptions of their usage. A lot of the content is (intentionally) similar to FES and TMwR; the goal of this vignette is to have documentation _in the package_ about these functions and why you'd use them, to improve discoverability.

This PR also adds a little bit to the `group_vfold_cv()` documentation to fix #347. It felt better to do this all at once rather than across two PRs.

This PR fixes #347 
This PR fixes #274

### CODE REVIEW

File: vignettes/Common_Patterns.Rmd [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = rlang::is_installed("modeldata")
+)
+```
+
+rsample provides a number of resampling methods which are broadly applicable to a wide variety of modeling applications. This vignette walks through the most popular methods in the package, with brief descriptions of how they can be applied. For a more in-depth overview of resampling, check out the matching chapters in [Tidy Modeling with R](https://www.tmwr.org/resampling.html) and [Feature Engineering and Selection](http://www.feat.engineering/resampling.html).
```
Comment (hfrick) [inline vignettes/Common_Patterns.Rmd:18]:
```suggestion
The rsample package provides a number of resampling methods which are broadly applicable to a wide variety of modeling applications. This vignette walks through the most popular methods in the package, with brief descriptions of how they can be applied. For a more in-depth overview of resampling, check out the matching chapters in [Tidy Modeling with R](https://www.tmwr.org/resampling.html) and [Feature Engineering and Selection](http://www.feat.engineering/resampling.html).
```

File: vignettes/Common_Patterns.Rmd [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+Let's go ahead and load rsample now:
+
+```{r setup}
+library(rsample)
+```
+
+As well as dplyr, for the pipe operator `%>%`:
```
Comment (hfrick) [inline vignettes/Common_Patterns.Rmd:26]: Do we use any other dplyr functionality? if not, we could just load magrittr

File: vignettes/Common_Patterns.Rmd [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+mc_cv(ames, prop = 0.8, times = 2)
+```
+
+Just as with `validation_set()`, you can control the proportion of your data assigned to the analysis fold using `prop`. You can also control the number of resamples you create using the `times` argument.
+
+Monte-Carlo cross-validation tends to produce more biased estimates than V-fold. As such, when computationally feasible we typically recommend using [five or so repeats of 10-fold cross-validation](https://bookdown.org/max/FES/resampling.html#resample-var-bias) for model assessment.
+
+### Bootstrap Resampling
+
+The last primary resampling function in rsample is `bootstraps()`, for bootstrap resampling. A "bootstrap sample" is a sample of your data set, the same size as your data set, taken with replacement so that a single observation might be sampled multiple times. The assessment set is then made up of all the observations that weren't selected for the analysis set. Generally, bootstrap resampling produces pessimistic estimates of model accuracy.
```
Comment (hfrick) [inline vignettes/Common_Patterns.Rmd:130]:
```suggestion
The last primary resampling technique in rsample is bootstrap resampling. A "bootstrap sample" is a sample of your data set, the same size as your data set, taken with replacement so that a single observation might be sampled multiple times. The assessment set is then made up of all the observations that weren't selected for the analysis set. Generally, bootstrap resampling produces pessimistic estimates of model accuracy.
```

File: vignettes/Common_Patterns.Rmd [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+As mentioned, you can create bootstrap resamples in rsample using the `bootstraps()` function. While you can't control the proportion of data in each set -- the assessment set of a bootstrap resample is always the same size as the training data -- the function otherwise works exactly like `mc_cv()`:
```
Comment (hfrick) [inline vignettes/Common_Patterns.Rmd:132]:
```suggestion
You can create bootstrap resamples in rsample using the `bootstraps()` function. While you can't control the proportion of data in each set -- the assessment set of a bootstrap resample is always the same size as the training data -- the function otherwise works exactly like `mc_cv()`:
```

File: vignettes/Common_Patterns.Rmd [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+```
+
+## Grouped Resampling
+
+Often, some observations in your data will be "more related" to each other than would be probable under random chance, for instance because they represent repeated measurements of the same subject or were all collected at a single location. In these situations, you often want to assign all related observations to either the analysis or assessment fold as a group, to avoid having assessment data that's closely related to the data used to fit a model.
+
+All of the functions discussed so far have a "grouped resampling" variation to handle these situations. These functions all start with the `group_` prefix, and use the argument `group` to specify which column should be used to group observations. Other than respecting these groups, these functions all work like their ungrouped variants:
+
+```{r}
+resample <- group_initial_split(Orange, group = Tree)
```
Comment (hfrick) [inline vignettes/Common_Patterns.Rmd:161]: oh, where is this dataset coming from?

**Review by hfrick — APPROVED** [review]
Really nice! 🙌 

Where is the `Orange` dataset coming from? That had me surprised.

**Comment by mikemahoney218** [comment]
It's one of the base R data sets (repeated measurements of five orange trees) -- I wanted something with fewer levels than Ames to demonstrate grouping. Should I note that it's from base, or change to something more famous (mtcars)?

File: vignettes/Common_Patterns.Rmd (hunk 2 above)
Comment (mikemahoney218) [reply vignettes/Common_Patterns.Rmd:26]: dplyr is already a package dependency, while magrittr "isn't" (only brought in recursively) -- so I could add magrittr as a dependency, or just stick with dplyr here. At the moment this is a vignette, not an article, so we can't use website dependencies for it either.

Comment (hfrick) [reply vignettes/Common_Patterns.Rmd:26]: ah, right, makes sense 👍

**Comment by mikemahoney218** [comment]
As Gabor mentioned this morning, R-Devel is busted today:
https://github.com/wch/r-source/commit/ef0a4a211f92640e71e87cbe7b66323f2c239e4d

So, not much to be done about that on our end.
